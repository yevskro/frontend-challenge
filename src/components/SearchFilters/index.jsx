import React from 'react';
import styled from 'styled-components';

function SearchFilters({ children, flexEnd }) {
  return (
    <StyledSearchFilters flexEnd={flexEnd} role="listbox">
      {children}
    </StyledSearchFilters>
  );
}

const StyledSearchFilters = styled.ul`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  justify-content: ${(props) => (props.flexEnd ? 'flex-end' : 'flex-start')};
  @media (max-width: 415px) {
    justify-content: flex-start;
  }
`;

export default SearchFilters;
