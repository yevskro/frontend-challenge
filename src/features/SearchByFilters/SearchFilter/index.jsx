import React from 'react';
import styled from 'styled-components';

function SearchFilter({
  children,
  filter,
  remove,
  onRemoveFilter,
  onAddFilter,
}) {
  /* a filter that is removeable should only handle an onClick
  on the removing icon.
    a filter that is addable shall not have the remove prop or
    the prop would be set to false. this filter can be clickable
    by the text content
  */
  return (
    <StyledSearchFilter>
      <Text
        onClick={remove ? () => {} : () => onAddFilter(filter)}
        remove={remove}
      >
        {children}
      </Text>
      {remove ? (
        <Remove onClick={() => onRemoveFilter(filter)}>
          <img src="/assets/images/icon-remove.svg" alt="remove icon" />
        </Remove>
      ) : (
        ''
      )}
    </StyledSearchFilter>
  );
}

const StyledSearchFilter = styled.li`
  display: flex;
  justify-content: center;
  align-items: center;
  height: 32px;
  margin: 10px 16px 10px 0px;
  border-radius: 5px;
  overflow: hidden; /* hides the squared border on hover of the span child */
  background-color: ${({ theme }) =>
    theme.color.neutral.lightGrayishCyanFilterTablets};

  @media (min-width: 415px) {
    margin: 10px 0px 10px 16px;
  }
`;

const Text = styled.span`
  display: flex;
  align-items: center;
  height: 100%;
  padding: 0px 9px 0px 8px;
  color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};
  font-size: ${({ theme }) => theme.fontSize.small};
  font-weight: ${({ theme }) => theme.fontWeight.bold};

  &:hover {
    /* only apply colors if this is a not removable filter */
    color: ${({ remove, theme }) =>
      remove ? theme.color.primary.desaturtedDarkCyan : 'white'};
    background-color: ${({ remove, theme }) =>
      remove
        ? theme.color.neutral.lightGrayishCyanFilterTablets
        : theme.color.primary.desaturtedDarkCyan};
    cursor: ${({ remove }) => (remove ? 'auto' : 'pointer')};
  }
`;

const Remove = styled.div`
  display: flex;
  justify-content: center;
  align-items: center;
  width: 32px;
  height: 32px;
  border-radius: 0px 5px 5px 0px;
  background-color: ${({ theme }) => theme.color.primary.desaturtedDarkCyan};

  &:hover {
    background-color: ${({ theme }) => theme.color.neutral.veryDarkGrayishCyan};
    cursor: pointer;
  }
`;

export default SearchFilter;
