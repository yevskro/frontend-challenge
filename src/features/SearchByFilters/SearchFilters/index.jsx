import React from 'react';
import styled from 'styled-components';

/* search filters has the ability to flex its children either to flex-end
or flex-start */
function SearchFilters({ children, flexEnd }) {
  return (
    <StyledSearchFilters flexEnd={flexEnd}>{children}</StyledSearchFilters>
  );
}

const StyledSearchFilters = styled.ul`
  display: flex;
  flex-wrap: wrap;
  width: 100%;
  justify-content: ${(props) => (props.flexEnd ? 'flex-end' : 'flex-start')};
  @media (max-width: 415px) {
    justify-content: flex-start;
  }
`;

export default SearchFilters;
