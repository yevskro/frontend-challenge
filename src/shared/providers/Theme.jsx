import React from 'react';
import { ThemeProvider } from 'styled-components';
/* theme developed based on the style-guide */
const theme = {
  color: {
    primary: {
      desaturtedDarkCyan: 'hsl(180, 29%, 50%)',
    },
    neutral: {
      lightGrayishCyanBackground: 'hsl(180, 52%, 96%)',
      lightGrayishCyanFilterTablets: 'hsl(180, 31%, 95%)',
      darkGrayishCyan: 'hsl(180, 8%, 52%)',
      veryDarkGrayishCyan: 'hsl(180, 14%, 20%)',
    },
    shadow: 'rgba(94, 140, 160, 0.15)',
  },
  fontSize: {
    xxsmall: '0.7rem',
    xsmall: '0.8rem',
    small: '0.9rem',
    medium: '1rem',
    large: '1.2rem',
  },
  fontWeight: {
    thin: '400',
    normal: '500',
    bold: '600',
  },
  fontFamily: '"Spartan", "sans-serif"',
};

const Theme = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);

export default Theme;
