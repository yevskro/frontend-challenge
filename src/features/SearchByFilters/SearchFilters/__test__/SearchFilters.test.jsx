import React from 'react';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Theme from '../../../../shared/providers/Theme';
import SearchFilters from '..';
import SearchFilter from '../../SearchFilter';

test('renders', () => {
  render(
    <Theme>
      <SearchFilters>
        <SearchFilter>CSS</SearchFilter>
      </SearchFilters>
    </Theme>
  );

  expect(screen.getByText('CSS')).toBeInTheDocument();
});

test('component default justification is to flex-start', () => {
  render(
    <Theme>
      <SearchFilters>
        <SearchFilter>CSS</SearchFilter>
      </SearchFilters>
    </Theme>
  );

  expect(screen.getByRole('list')).toHaveStyleRule(
    'justify-content',
    'flex-start'
  );
});

test('component can adjust its justification to flex-end', () => {
  render(
    <Theme>
      <SearchFilters flexEnd>
        <SearchFilter>CSS</SearchFilter>
      </SearchFilters>
    </Theme>
  );

  expect(screen.getByRole('list')).toHaveStyleRule(
    'justify-content',
    'flex-end'
  );
});

test('snapshot', async () => {
  const tree = renderer
    .create(
      <Theme>
        <SearchFilters flexEnd>
          <SearchFilter>CSS</SearchFilter>
        </SearchFilters>
        <SearchFilters>
          <SearchFilter>CSS</SearchFilter>
        </SearchFilters>
      </Theme>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
