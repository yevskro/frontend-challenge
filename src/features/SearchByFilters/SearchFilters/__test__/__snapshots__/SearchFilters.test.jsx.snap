// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  margin: 10px 16px 10px 0px;
  border-radius: 5px;
  overflow: hidden;
  background-color: hsl(180,31%,95%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  padding: 0px 9px 0px 8px;
  color: hsl(180,29%,50%);
  font-size: 0.9rem;
  font-weight: 600;
}

.c2:hover {
  color: white;
  background-color: hsl(180,29%,50%);
  cursor: pointer;
}

@media (max-width:415px) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (max-width:415px) {

}

@media (min-width:415px) {
  .c1 {
    margin: 10px 0px 10px 16px;
  }
}

<ul
    className="c0"
  >
    <li
      className="c1"
    >
      <span
        className="c2"
        onClick={[Function]}
      >
        CSS
      </span>
      
    </li>
  </ul>,
  .c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 32px;
  margin: 10px 16px 10px 0px;
  border-radius: 5px;
  overflow: hidden;
  background-color: hsl(180,31%,95%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
  padding: 0px 9px 0px 8px;
  color: hsl(180,29%,50%);
  font-size: 0.9rem;
  font-weight: 600;
}

.c2:hover {
  color: white;
  background-color: hsl(180,29%,50%);
  cursor: pointer;
}

@media (max-width:415px) {

}

@media (max-width:415px) {
  .c0 {
    -webkit-box-pack: start;
    -webkit-justify-content: flex-start;
    -ms-flex-pack: start;
    justify-content: flex-start;
  }
}

@media (min-width:415px) {
  .c1 {
    margin: 10px 0px 10px 16px;
  }
}

<ul
    className="c0"
  >
    <li
      className="c1"
    >
      <span
        className="c2"
        onClick={[Function]}
      >
        CSS
      </span>
      
    </li>
  </ul>,
]
`;
